### DISCUSSION TITLE
Hardcoding strings such as "http.response.start", "type" vs Constants/Enums

### DISCUSSION THREAD

**Discussion by ggsggs** [encode/starlette#2307] [Q&A] [answered]
Hello!

Thanks for the great work.

I am implementing a Pure ASGI middleware looking at the [docs ](https://www.starlette.io/middleware/#inspecting-or-modifying-the-response) and I was wondering why there isn't a module for commonly constant strings used in comparisons (`message["type"] == "http.response.start"`) or for commonly used keys in Mappings/dicts  (`scope["type"]`).

What I mean, as an example. Use some sort of `starlette.ASGIEvents.types.HTTPRESPONSESTART`  instead of a hardcoded `"http.response.start"`.

Thank you.

**Comment by Kludex** [comment, marked as answer]
It is how it is.

**Reply by ggsggs** [reply]
Well, I can see that. Would that be an "improvement" if I helped with that or keeping it "lean" is more important?

**Reply by Kludex** [reply]
I'd prefer to keep it as is.

**Reply by ggsggs** [reply]
Okay, fair enough. Thank you!
